.github/workflows/publish-branch.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish branch
+on:
+  push: # run on push to any branch except master
+    branches:
+      - "!master"
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+
+      - uses: ./.github/actions/setup
+
+      - name: Set version to include branch
+        run: |
+          # Get branch name and replace / with -
+          BRANCH_NAME=$(echo "${GITHUB_REF#refs/heads}" | sed -e "s/\//-/g")
+          sed -i '/version=/ s/-SNAPSHOT/-'"$BRANCH_NAME"'-SNAPSHOT/g' ./gradle.properties
```

Comment:
One potential changes will come soon is the current master branch will be renamed to v1.x/master soon, and v2.x/master will be changed to the default branch of the repo. this may affect your branches check like:

```
push:
    branches:
      - master
```

When we modifying the triggering, can also think how we can modify the branch name/version modification, and maybe potentially merge `.github/workflows/publish-branch.yml` and `.github/workflows/snapshot-release.yml`.